File: src/main/java/net/hectus/neobb/turn/person_game/categorization/ArmorCategory.java
package net.hectus.neobb.turn.person_game.categorization;

import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextColor;
import org.bukkit.Material;

public interface ArmorCategory extends AttackCategory {
    @Override
    default TextColor categoryColor() {
        return NamedTextColor.AQUA;
    }

    @Override
    default Material categoryItem() {
        return Material.OAK_DOOR;
    }

    @Override
    default String categoryName() {
        return "armor";
    }

    int armor();
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/person_game/categorization/ArmorCategory.kt
package net.hectus.neobb.modes.turn.person_game.categorization

import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextColor
import org.bukkit.Material

interface ArmorCategory : AttackCategory {
    override val categoryColor: TextColor get() = NamedTextColor.AQUA

    override val categoryItem: Material get() = Material.OAK_DOOR
    override val categoryName: String get() = "armor"

    fun armor(): Int
}
